app/javascript/components/visualization/ScatterPlot.jsx

```diff
@@ -941,3 +923,71 @@ function getDragMode(isCellSelecting) {
 }
 
 ScatterPlot.getPlotlyTraces = getPlotlyTraces
+
+/** Intersect filtered cells with dataArrays results */
+export function intersect(filteredCells, scatter) {
+  const intersectedData = {}
+  const dataArrays = scatter.data
+  const keys = Object.keys(dataArrays)
+  keys.forEach(key => intersectedData[key] = [])
+  // Uncomment for debugging, but be sure to comment out before PR
+  // window.dataArrays = dataArrays
+
+  // array of cell indexes that are being plotted
+  const plotted = []
+  for (let i = 0; i < filteredCells.length; i++) {
+    const filteredCell = filteredCells[i]
+    const allCellsIndex = filteredCell.allCellsIndex
+    plotted.push(allCellsIndex)
+    for (let j = 0; j < keys.length; j++) {
+      const key = keys[j]
+      const dataArray = dataArrays[key]
+      const filteredElement = dataArray[allCellsIndex]
+      intersectedData[key].push(filteredElement)
+    }
+  }
+  return [intersectedData, plotted]
+}
+
+ScatterPlot.intersect = intersect
+
+/**
+ * Reassign plot data to change cells filtered by cell faceting to --Filtered-- trace (light grey, transparent)
+ * this keeps the shape of the original plot but clearly shows what cells are excluded
+ *
+ * @param plotted {Array} Indices of points that are to be plotted
+ * @param originalData {Object} original scatter data object
+ * @param filteredData {Object} filtered scatter data object from cell faceting
+ * @returns {Object} reorganized scatter data object with new --Filtered-- trace
+ */
+export function reassignFilteredCells(plotted, originalData, filteredData) {
+  const allIndexes = [...Array(originalData['x'].length).keys()]
+  const plottedSet = new Set(plotted)
+  const reassignedIndices = allIndexes.filter(i => !plottedSet.has(i))
```

Comment:
These couple of lines seem a bit unnecessarily complex. It took me a while to sort out what it was doing, and it's technically making 3 extra arrays that aren't really needed (via the spread, the .keys() and the [] brackets) I think this could be accomplished using just a forloop like:

`const reassignedIndices = []
 const plottedSet = new Set(plotted)
for (let i = 0;  i < originalData['x'].length; i++)
    if (!plottedSet.has(i)) reassignedIndices.push(i)`

Reply:
Oh that's a great idea - let me do some benchmarking to see how that performs but I have a feeling that will be both simpler and faster.

Reply:
Yup - twice as fast and definitely easier to read.  FWIW this has the same performance as the earlier comment (only about 120ms vs 60) but the readability improvement alone is worth it.